core/src/main/java/hudson/slaves/SimpleScheduledRetentionStrategy.java

```diff
@@ -239,7 +239,7 @@ public void run() {
         return 1;
     }
 
-    private boolean isOnlineScheduled() {
+    private synchronized boolean isOnlineScheduled() {
```

Comment:
Everything else that manipulates `lastStart`, `lastStop`, etc. is already synchronized, so this should be synchronized as well for consistency. `updateStartStopWindow` is already synchronized, so it doesn't hurt to synchronize on the whole method.